src/app/client/src/app/modules/learn/components/course-consumption/assessment-player/assessment-player.component.ts

```diff
@@ -541,10 +543,11 @@ export class AssessmentPlayerComponent implements OnInit, OnDestroy, ComponentCa
           unit.isUnitConsumed = consumedContent.length === 1;
           unit.progress = consumedContent.length ? 100 : 0;
           unit.isUnitConsumptionStart = Boolean(consumedContent.length);
+          if(unit?.relationalMetadata?.optional=== false) {
```

Comment:
@Saurabhbalke please describe the code in a comment

Reply:
hii @princegupta1131  sir,
added comments on the changes, PTAL!
thanks